File: tests/go/cases/case_223_shadowed_sink_local_02.go
package escape_tests

var retainedCase223 = []map[string]string{}

func Case223ShadowedSinkLocal02(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "shadowed_sink_local_02",
		"entity": "extreme",
		"stage": "stress",
		"input": raw,
		"checksum": "shadowed_sink_local_02:" + raw,
	}
	retainedCase := []map[string]string{}
	retainedCase = append(retainedCase, payload)
	// SAFE: local shadow variable only.
	return payload["checksum"]
}

File: tests/nodejs/cases/case_223_shadowed_sink_local_02.js
/**
 * Task case 223: shadowed_sink_local_02 deep stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case223ShadowedSinkLocal02(input) {
  const taskName = 'shadowed_sink_local_02';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'extreme',
    stage: 'stress',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const retainedCache = new Map();
  retainedCache.set('tmp', payload);
  // SAFE: local shadow hides global sink symbol.
  return payload.checksum;
}

module.exports = { case223ShadowedSinkLocal02 };
